File: backend/app/schemas/small_win.py
import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class SmallWinCreate(BaseModel):
    date: datetime.date
    text: str = Field(min_length=1, max_length=500)
    entry_type: str = "win"
    completed: bool | None = None
    category: str | None = None


class SmallWinUpdate(BaseModel):
    text: str | None = Field(None, min_length=1, max_length=500)
    completed: bool | None = None
    category: str | None = None


class SmallWinResponse(BaseModel):
    id: UUID
    date: datetime.date
    text: str
    entry_type: str
    completed: bool | None
    category: str | None
    created_at: datetime.datetime
    updated_at: datetime.datetime

    model_config = ConfigDict(from_attributes=True)


class DayCount(BaseModel):
    date: datetime.date
    count: int


class SmallWinStats(BaseModel):
    total_wins: int
    wins_last_7_days: list[DayCount]

File: frontend/types/small-win.ts
export type SmallWinCategory = "Work" | "Health" | "Personal Growth" | "General"

export interface SmallWinResponse {
  id: string
  date: string
  text: string
  entry_type: "win" | "task"
  completed: boolean | null
  category?: SmallWinCategory | null
  created_at: string
  updated_at: string
}

export interface SmallWinCreateInput {
  date: string
  text: string
  entry_type?: "win" | "task"
  category?: SmallWinCategory | null
}

export interface SmallWinUpdateInput {
  text?: string
  completed?: boolean
  category?: SmallWinCategory | null
}

export interface DayCount {
  date: string
  count: number
}

export interface SmallWinStats {
  total_wins: number
  wins_last_7_days: DayCount[]
}
